### PR TITLE
[Hold] Add the needed table columns for client Oauth.

### PR DESCRIPTION
**Pull request by sgoodwin**
This is to address issue #4174.

### CODE REVIEW

**Comment by ErisDS** [comment]
Hi @sgoodwin thanks for this :) I've added 'HOLD' to the title of this PR as I want to get the next release out before merging it. As soon as we release 0.5.8, there's another schema change issue (#4687) we need to do, and if we merge them together we can do a single version bump & migrating release.

The next release should be happening this coming week, so the hold will only be brief.

**Comment by sgoodwin** [comment]
No problem, there you go.

**Comment by jaswilli** [comment]
Not at all a comment on the quality of this PR but we should hold this until the features that require it are under development--it's tough to know exactly what database changes are needed until we get into it.

**Comment by sgoodwin** [comment]
That's fair. What can I do to help the other issues along? I have zero ember experience and very little node experience. I'm surprised I figured this much out.

**Comment by ErisDS** [comment]
One major blocker is that OAuth itself is a fairly tricky subject (let alone the programming around it) and our resident expert @sebgie has been working on other projects. We're also a bit stuck without a way to [migrate permissions](https://github.com/TryGhost/Ghost/issues/3910). If you follow along with [meetings / or write-ups](http://dev.ghost.org/tag/meetings/) you'll see we're planning to shift the team's focus from user-facing features as things are at the moment, towards the developer-facing features like this once we've shipped a few more of the features we're working on. 

In short, you can help us out in a number of ways, contributing in any way to the project (testing pull requests, fixing bugs and smaller issues) helps us move faster and get to this stuff sooner ;)

**Comment by ErisDS** [comment]
Going to close this, as the commit has been pulled into #5706 - a full set of migrations for 004. It will be merged shortly.
